### PR TITLE
Add Search::Results::Service object

### PR DESCRIPTION
**Pull request by kfrz**
## Background

As part of the brand consolidation, we are expanding our existing search functionality.  We will now be integrating the Search.gov search results into a new page within our existing site.

#### Originating Issue
https://github.com/department-of-veterans-affairs/vets.gov-team/issues/12651#issuecomment-416662196

This PR adds a service object class, transaction response class, and spec coverage. 

#### Unique to this PR
- [x] Create a [service class and object](https://github.com/department-of-veterans-affairs/vets.gov-team/blob/master/Work%20Practices/Engineering/vets-api-endpoint-punch-list.md#service-object) wrapping the [Search.gov API search endpoint](https://search.usa.gov/sites/6277/api_instructions)
- [x] ~~Rate limiting considerations, if applicable (TBD)~~
- [x] Spec coverage

#### Applies to all PRs

- [x] Appropriate logging
- [x] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [x] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: lib/search/service.rb [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    def access_key
+      Settings.search.access_key
+    end
+
+    def handle_error(error)
+      case error
+      when Common::Client::Errors::ClientError
+        message = parse_messages(error).first
+        log_error_message(message)
+        # raise_backend_exception('SEARCH_400', self.class, error) if message.present?
```
Comment (kfrz) [inline lib/search/service.rb]: @hpjaj do you think we want to raise a backend exception for a given `400` level status? Or just pass through the message to the response class? 

I'm trying to coerce the search.gov api into giving me some 500 level error codes but have had no luck, and as we know their docs have zero examples... but those are the codes I'd imagine want an exception raised. Thoughts?

File: lib/search/service.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require 'common/client/base'
+require 'search/response'
+
+module Search
+  class Service < Common::Client::Base
```
Comment (hpjaj) [inline lib/search/service.rb]: Can you pls add in some Yard docs explaining that this is wrapper for the Search.gov API, with links out to their API docs.

File: lib/search/response.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    def self.from(response)
+      body = response.body
+      new(response.status, body: body)
+    end
+
+    def ok?
+      status == 200
+    end
+
+    def cache?
```
Comment (dhall) [inline lib/search/response.rb]: curious about the dup of ok? and cache? are both needed for context .. would alias make more sense

Comment (hpjaj) [reply lib/search/response.rb]: Yeah, this sets up the API to be cachable with our internal framework.  Good point.  I believe framework has moved away from the `ok?` method in favor of the `cache?` method, so we could just move the `status == 200` into `cache?` and remove `ok?`.

File: lib/search/configuration.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        faraday.use      Faraday::Response::RaiseError
+
+        faraday.response :betamocks if mock_enabled?
+        faraday.response :snakecase, symbolize: false
+        faraday.response :json, content_type: /\bjson/ # ensures only json content types parsed
+        faraday.adapter Faraday.default_adapter
+      end
+    end
+
+    def mock_enabled?
```
[the remark this answers is not in the document]
Comment (hpjaj) [reply lib/search/configuration.rb]: Yeah, this is the precedence we have been following for all of the 3rd party services that we have integrated, that we use betamocks with.  We'll be adding betamock support to this service in an upcoming ticket.

**Pull request merged**